src/Akeneo/Category/back/Infrastructure/Storage/Save/Query/SqlUpsertCategoryBase.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\Infrastructure\Storage\Save\Query;
+
+use Akeneo\Category\Application\Storage\Save\Query\UpsertCategoryBase;
+use Akeneo\Category\Domain\Model\Category;
+use Akeneo\Category\Domain\ValueObject\CategoryId;
+use Akeneo\Category\Domain\ValueObject\Code;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception;
+
+/**
+ * Save values from model into pim_catolog_category table:
+ * The values are inserted if the id is new, they are updated if the id already exists.
+ *
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class SqlUpsertCategoryBase implements UpsertCategoryBase
```

Comment:
As per already exist 
```suggestion
class UpsertCategoryBaseSql implements UpsertCategoryBase
```

Reply:
done